app/components/participant-creator/component.js

```diff
@@ -99,28 +101,42 @@ export default Ember.Component.extend({
         },
         addExtraField() {
             var next = this.get('nextExtra');
-            this.get('extra').pushObject({
-                key: next,
-                value: null
-            });
+            var fieldExists = false;
+            // Do not allow users to add duplicate studyId field
+            if (next === 'studyId') {
+                fieldExists = true;
+            }
+            for (var item of this.get('extra')) {
```

Comment:
this.get(`extra`) is a list of object of the format {key: <fieldname>, value: <value>} so it's not possible to check `Object.keys(extra)` unless I reformat it into an object first.